ci-operator/step-registry/hypershift/aws/setup-nested-management-cluster/hypershift-aws-setup-nested-management-cluster-chain.yaml

```diff
@@ -59,10 +59,18 @@ chain:
 
       # Data for cluster bot.
       # The kubeadmin-password secret is reconciled only after the kas is available so giving a few seconds to avoid racing and move forward otherwise. 
-      sleep 15
-      echo "Getting ${CLUSTER_NAME}-kubeadmin-password from namespace ${HYPERSHIFT_NAMESPACE}"
-      oc get secret -n${HYPERSHIFT_NAMESPACE} ${CLUSTER_NAME}-kubeadmin-password --template={{.data.password}} | base64 -d > ${SHARED_DIR}/kubeadmin-password
-
+      timeout --foreground 5m bash <<- "EOF"
+        while [[ -z "$(oc get --namespace=${HYPERSHIFT_NAMESPACE} hostedcluster ${CLUSTER_NAME} -o jsonpath='{.status.kubeadminPassword.name}')" ]]; do
+          echo "Waiting for kubeadmin password to be populated"
+          sleep 5
+        done
+        KUBEADMIN_SECRET_NAME=$(oc get hostedcluster --namespace=${HYPERSHIFT_NAMESPACE} ${CLUSTER_NAME} -o jsonpath='{.status.kubeadminPassword.name}')
+        while ! KUBEADMIN_SECRET_ENCODED=$(oc get secret ${KUBEADMIN_SECRET_NAME} -n${HYPERSHIFT_NAMESPACE}  --template={{.data.password}}); do
+          echo "Kubeadmin secret does not exist. Waiting for 5 seconds"
+          sleep 5
```

Comment:
why is this sleep needed given we already check status above this loop shouldn't be needed